docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+
+Note: Detailed list of pros/cons of using Tailwind are listed in this [ADR](https://github.com/mozilla/fxa/blob/main/docs/adr/0018-use-tailwind-with-custom-scss.md#option-b---use-an-existing-utility-library)
+
+### Option C - Use CSS-in-JS
+
+- Description
+  - CSS-in-JS is a styling technique that leverages the power of JavaScript to style components. There are multiple implementations of this concept in the form of libraries such as [Styled Components](https://styled-components.com/), [Emotion](https://emotion.sh/docs/introduction), [JSS](https://cssinjs.org/).
+- Pros
+  - This technique offers modularity which means that we do not have to maintain a bunch of stylesheets as CSS-in-JS abstracts the CSS model to the component level, rather than the document level.
+  - These libraries offer full CSS support including features like media queries which was not possible with option A.
+  - Most of the libraries support server-side rendering, making them eligible for use with our React option.
+  - Components are reusable, so we will only have to write them once, then use them everywhere.
+  - All CSS-in-JS libraries generate unique CSS class names and because of this feature, we never have to worry about CSS class name collisions, or wasted time spent coming up with unique class names across the entire codebase.
+  - A large number of companies use CSS-in-JS in production, including Reddit, Patreon, Target, Atlassian, Vogue, GitHub, Coinbase, and many more.
+- Cons
+  - Putting a CSS-in-JS library into use will add an extra layer of dependencies to our front-end stack, although that could be mitigated by simply using CSS-in-JS files and not any library if we're using React.
+  - CSS-in-JS libraries generate selectors automatically which might worsen code readability, and can be difficult for new contributors.
+  - We haven't employed this technique before in any of our fxa services
+
+### Option D - Use a CSS pre-processor like SASS
+
+- Description
+  - [SASS](https://sass-lang.com/documentation) (which stands for 'Syntactically awesome style sheets') is an extension of CSS that enables making use of variables, nested rules, inline imports and other interesting functionalities that can make writing CSS much more powerful. in a stylesheet. It comes with 2 kinds of syntax: SCSS (Sassy CSS) and Indented (simply called 'Sass'). SCSS files use .scss extension and are fully compliant with CSS syntax, whereas the SASS files are not and hence, we will be discussing only about using .scss files as our styling solution in this section.
+- Pros
+  - Sass offers little-to-no learning curve especially if we're using SCSS syntax since it's nothing but CSS.
+  - It has large community and well documented.
+  - SASS offers modularization. With the help of `@import` directive, we can import partials (small SASS files) into other SASS files.
+  - It is compatible with all versions of CSS so any available CSS libraries can be employed with SASS.
+- Cons
```

Comment:
A con in my eyes is that we use a Tailwind config file for color palettes and common design values, and if we were to use SASS it'd likely involve setting up a new `variables.scss` or similar where we'd set up new color variables, etc, and this means to some extent we'd want to keep it in sync with the Tailwind config.